src/backend/renderer/utils.rs

```diff
@@ -156,27 +157,16 @@ impl RendererSurfaceState {
         self.buffer.as_ref()
     }
 
-    /// Location of the new buffer relative to the previous one
-    ///
-    /// The x and y arguments specify the location of the new pending buffer's upper left corner,
-    /// relative to the current buffer's upper left corner, in surface-local coordinates.
+    /// Location of the buffer relative to the previous call of take_accumulated_buffer_delta
     ///
     /// In other words, the x and y, combined with the new surface size define in which directions
-    /// the surface's size changes.
-    pub fn buffer_delta(&self) -> Option<Point<i32, Logical>> {
-        self.buffer_delta
-    }
-
-    /// Same as [`Self::buffer_delta`] but takes the delta.
+    /// the surface's size changed since last call to this method.
     ///
     /// Once delta is taken this method returns `None` to avoid processing it several times.
-    pub fn take_buffer_delta(&mut self) -> Option<Point<i32, Logical>> {
-        if self.buffer_delta_ack {
-            None
-        } else {
-            self.buffer_delta_ack = true;
-            self.buffer_delta
-        }
+    pub fn take_accumulated_buffer_delta(&mut self) -> Point<i32, Logical> {
+        let accumulated_buffer_delta = self.accumulated_buffer_delta;
+        self.accumulated_buffer_delta = (0, 0).into();
+        accumulated_buffer_delta
```

Comment:
```suggestion
        std::mem::replace(&mut self.accumulated_buffer_delta, (0, 0).into())
```

Reply:
Actually, I think even `std::mem::take(&mut self.accumulated_buffer_delta)` would work, thanks to the `Default` impl of `Point`.

Reply:
Done using `std::mem::take`